scripts/compare_books.py

```diff
@@ -140,18 +151,21 @@ def problem_tokenizer(file_string):
         if match_str is not "":
             yield Token(token_type, match_str)
 
-# Must consider: \begin{enumerate}, \end{enumerate}, \item, \setcounter{...}{...}, \begin{outer_problem}, \begin{inner_problem}, \begin{iinner_problem}
+
+# Must consider: \begin{enumerate}, \end{enumerate}, \item, \setcounter{...}{...}, \begin{outer_problem},
+# \begin{inner_problem}, \begin{iinner_problem}
 def problem_generator(file_string, is_answer_key=False):
     global counters
 
+    # no idea what's supposed to be local and what's supposed to shadow, so am just leaving it like this
```

Comment:
@anematode

Reply:
I am assuming that the one in the beginning is a declaration and the ones later in the if are an assignment?
```py
def problem_generator(file_string, is_answer_key=False):
    global counters

    enum_depth = 0  # declaration?
    record_problem_n = 0

    set_counter_by_value("outer", 0)
    set_counter_by_value("inner", 0)
    set_counter_by_value("iinner", 0)

    for x in range(1, 4):
        set_counter_by_value("enum" + "i" * x, 0)

    for token in problem_tokenizer(file_string):
        if token.type == "begin_enumerate":
            name = token.get_enumeration_name()

            if name == "enumerate":
                enum_depth += 1  # assignment?

                set_counter_by_value("enum" + "i" * enum_depth, 0)
            else:
                if name == "outer_problem":
                    enum_depth = 1  # assignment?
```
If the above assumption is true, the assignments in the else of the if are not properly reassigning enum_depth to their values.

Reply:
Screenshot showing that the later ones are assignments
<img width="343" alt="screenshot" src="https://user-images.githubusercontent.com/60120929/108296863-2c07dd00-714f-11eb-9311-9b99c5b06946.png">